netket/graph/space_group.py

```diff
@@ -69,6 +69,14 @@ def product(p: Translation, q: Translation):
     return Translation(p(np.asarray(q)), p._vector + q._vector)
 
 
+def _ensure_iterable(x):
+    """Extracts iterables given in varargs"""
+    if isinstance(x[0], Iterable):
+        return x[0]
```

Comment:
Something like this should help (please double-check, I haven't tested this code):
```suggestion
    if isinstance(x[0], Iterable):
        if len(x) > 1:
            raise TypeError("This function requires either an array or several scalar arguments, but multiple arrays have been passed.")
        return x[0]
```

Reply:
Ah, you already did it. Nevermind.

Reply:
yep, I've just written the same thing myself, with a shorter error message